x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/ssl/SSLTrustRestrictionsTests.java

```diff
@@ -133,7 +132,8 @@ public Settings nodeSettings(int nodeOrdinal) {
 
     private void writeRestrictions(String trustedPattern) {
         try {
-            Files.write(restrictionsPath, Collections.singleton("trust.subject_name: \"" + trustedPattern + "\""));
+            Files.write(restrictionsTmpPath, Collections.singleton("trust.subject_name: \"" + trustedPattern + "\""));
+            Files.move(restrictionsTmpPath, restrictionsPath, StandardCopyOption.ATOMIC_MOVE);
```

Comment:
I think we should add `REPLACE_EXISTING` and handling of `AtomicMoveNotSupportedException`. `SecurityFiles` has code for this.